packages/back-end/src/models/GithubUserTokenModel.ts

```diff
@@ -0,0 +1,80 @@
+import { omit } from "lodash";
+import uniqid from "uniqid";
+import mongoose from "mongoose";
+import { refreshToken } from "@octokit/oauth-methods";
+import {
+  GithubUserTokenInterface,
+  CreateGithubUserTokenInput,
+} from "../../types/github";
+
+type GithubUserTokenDocument = mongoose.Document & GithubUserTokenInterface;
+
+const githubUserTokenSchema = new mongoose.Schema({
+  id: String,
+  token: String,
+  expiresAt: Date,
+  refreshToken: String,
+  refreshTokenExpiresAt: Date,
+  createdAt: Date,
+  updatedAt: Date,
+});
+
+const GithubUserTokenModel = mongoose.model<GithubUserTokenDocument>(
+  "GithubUserToken",
+  githubUserTokenSchema
+);
+
+const toInterface = (doc: GithubUserTokenDocument): GithubUserTokenInterface =>
+  omit(doc.toJSON<GithubUserTokenDocument>(), ["__v", "_id"]);
+
+export const createGithubUserToken = async (
+  token: CreateGithubUserTokenInput
+) => {
+  const doc = await GithubUserTokenModel.create({
+    ...token,
+    id: uniqid("ghut_"),
+    createdAt: new Date(),
+    updatedAt: new Date(),
+  });
+  return toInterface(doc);
+};
+
+export const doesTokenExist = async (tokenId: string) => {
+  return await GithubUserTokenModel.exists({
+    id: tokenId,
+    createdAt: {
+      $gte: new Date(new Date().getTime() - 1 * 60 * 1000), // 1 minute
```

Comment:
Good q - this was sort of leftover from the previous implementation - it was meant to prevent using tokens that were generated a long time ago, which could have been a vulnerability but with this new approach it isn't. Will clean it up